prediction_market_agent/db/pinecone_handler.py

```diff
@@ -4,36 +4,51 @@
 from datetime import datetime
 from typing import Optional
 
-from langchain_core.documents import Document
+from langchain_core.vectorstores import VectorStore
 from langchain_openai import OpenAIEmbeddings
 from langchain_pinecone import PineconeVectorStore
 from loguru import logger
-from pinecone import Pinecone
+from pinecone import Index, Pinecone
 from prediction_market_agent_tooling.markets.agent_market import FilterBy, SortBy
+from prediction_market_agent_tooling.markets.omen.data_models import OmenMarket
 from prediction_market_agent_tooling.markets.omen.omen_subgraph_handler import (
     OmenSubgraphHandler,
 )
+from pydantic.v1.types import SecretStr as SecretStrV1
 from tqdm import tqdm
 
 from prediction_market_agent.agents.think_thoroughly_agent.models import (
     PineconeMetadata,
 )
 from prediction_market_agent.utils import APIKeys
 
-INDEX_NAME = "omen-markets"
+# INDEX_NAME = "omen-markets"
+INDEX_NAME = "omen-index-text-embeddings-3-large"
 T = t.TypeVar("T")
 
 
 class PineconeHandler:
-    def __init__(self) -> None:
-        keys = APIKeys()
-        self.pc = Pinecone(api_key=keys.pinecone_api_key.get_secret_value())
-        self.index = self.pc.Index(INDEX_NAME)
+    vectorstore: VectorStore
```

Comment:
Could you give an explanation of the changes in the logic of this class that have improved the performance?

Reply:
Sure.
The problem was, I was chunking the texts (market titles) to insert (roughly 4,000 in buckets of 100). For every chunk, I was calling Pinecone and asking for the IDs there, and then doing a set(new_texts_to_add).difference(ids_already_in_db) -> https://github.com/gnosis/prediction-market-agent/pull/307/files#diff-708990d1fa3062d041926232998f54cd6a5116d1b02f18bb6a14aa520cae7894L58

Now I do the `set difference` in the beginning, before chunking. Once I removed markets already present in the DB, chunking can be done safely (also see function `deduplicate_markets` for the case where the same market_title occurs (we don't insert duplicates).

Reply:
Thanks. What is the reason for chunking here?

Reply:
Pinecone has very long latency when uploading too many texts, hence sending 100 every time is safer to not incur timeouts, retry, etc.